[Docs] Docs customization (#12700)
Sylius/Sylius: docs/customization/api.rst

        //...
    }

> Configuring endpoints using yaml
> ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

Reviewer: ```suggestion
Configuring endpoints using yaml
--------------------------------
```

> To remove endpoint from api all you need to do is specify which resource and operation
> should be removed in ``config/api_platform/config.yaml``.

Reviewer: ```suggestion
To remove an endpoint from the API using YAML you need to specify the operation of which resource
should be removed in ``config/api_platform/config.yaml``.
```

> If you want to remove for example admin endpoint ``GET`` from ``Zones`` you need to use ``(unset)`` key in yaml config.

Reviewer: ```suggestion
If you want to remove, for example, the admin ``GET`` endpoint of ``Zones``, you need to use the ``(unset)`` key in its yaml config.
```

.. code-block:: yaml
    '%sylius.model.zone.class%':
        collectionOperations:
            admin_get (unset): ~

> Using the ``(unset)`` key you can remove filters you dont need.

Reviewer: ```suggestion
Using the ``(unset)`` key you can also remove filters you don't need.
```

.. code-block:: yaml
    '%sylius.model.product.class%':
        collectionOperations:
            shop_get:
                filters (unset):
                    - app.product_new_filter

> If you need to add new filter in yaml configuration, simply place this code into ``config/api_platform/config.yaml``

Reviewer: ```suggestion
If you need to add a new filter in the yaml configuration, simply add this kind of code to ``config/api_platform/config.yaml``
```

.. code-block:: yaml
    '%sylius.model.product.class%':
        collectionOperations:
            shop_get:
                filters:
                    - app.product_new_filter

> To add new operation all you need to do is specify it in config file.

Reviewer: ```suggestion
To add a new operation, just specify it in the config file.
```

If you want to remove, for example, the admin ``GET`` endpoint of ``Zones``, you need to use the ``(unset)`` key in its yaml config.

.. code-block:: yaml

    '%sylius.model.zone.class%':
        collectionOperations:
            admin_get (unset): ~

Using the ``(unset)`` key you can also remove filters you don't need.

.. code-block:: yaml

    '%sylius.model.product.class%':
        collectionOperations:
            shop_get:
                filters (unset):
                    - app.product_new_filter

If you need to add a new filter in the yaml configuration, simply add this kind of code to ``config/api_platform/config.yaml``.

.. code-block:: yaml

    '%sylius.model.product.class%':
        collectionOperations:
            shop_get:
                filters:
                    - app.product_new_filter

To add a new operation, just specify it in the config file.

.. code-block:: yaml

    '%sylius.model.channel.class%':
        collectionOperations
            my_new_operation:
                method: GET
                path: /shop/channels
                normalization_context:
                    groups: ['shop:channel:read']

> 

Reviewer 2: IIRC you can also add a case with overwriting the entire endpoint configuration (you firstly should unset operation and after that add new one).

Author: good point, added a way to overwrite existing endpoint (we dont need to unset it)

    '%sylius.model.product.class%':
        collectionOperations:
            shop_get:
                filters (unset):
                    - app.product_new_filter

If you need to add a new filter in the yaml configuration, simply add this kind of code to ``config/api_platform/config.yaml``.

.. code-block:: yaml

    '%sylius.model.product.class%':
        collectionOperations:
            shop_get:
                filters:
                    - app.product_new_filter

To add a new operation, just specify it in the config file.

.. code-block:: yaml

    '%sylius.model.channel.class%':
        collectionOperations:
            my_new_operation:
                method: GET
                path: /shop/channels
                normalization_context:
                    groups: ['shop:channel:read']

You can also overwrite existing endpoints, for example let's change admin_get operation in order collectionOperations.

.. code-block:: yaml

    '%sylius.model.order.class%':
        collectionOperations:
            admin_get:
                path: /admin/orders/new_endpoint
                normalization_context:
                    groups: ['shop:channel:new_group']

> This way we can edit existing endpoint and add custom normalization or change path.

Reviewer 3: ```suggestion
This way we can edit the existing endpoint and add custom normalization or change the path.

```

> This way we can edit existing endpoint and add custom normalization or change path.

Reviewer 3: ```suggestion
This way we can edit the existing endpoint and add custom normalization or change the path.

```
